### PR TITLE
Add compression and remove incompatible plugins

### PR DESCRIPTION
**Pull request by derick-montague**
- Add gzip file compression

- Prefetch and preload plugins that add web loading primitives that are
not compatible with our server settings

- Move port to run dev from script flag to vue config file

- Remove sourcemap files from production build

### CODE REVIEW

**Review by dixsie — APPROVED** [review]

File: vue.config.js [hunk 1]
```diff
@@ -1,5 +1,24 @@
...
+      config.plugins.push(
+        new CompressionPlugin({
+          deleteOriginalAssets: true
+        })
+      );
+    }
+  },
+  chainWebpack: config => {
+    if (process.env.NODE_ENV === "production") {
+      config.plugins.delete("prefetch");
```
Comment (gtmills) [inline vue.config.js:20]: If we add something like `"prefetch-src 'self'; "` should be able to turn this back on. More of a note.

File: vue.config.js [hunk 2]
```diff
@@ -1,5 +1,24 @@
...
-    proxy: process.env.BASE_URL
+    proxy: process.env.BASE_URL,
+    port: 8000
+  },
+  productionSourceMap: false,
+  configureWebpack: config => {
+    if (process.env.NODE_ENV === "production") {
+      config.plugins.push(
+        new CompressionPlugin({
+          deleteOriginalAssets: true
```
Comment (gtmills) [inline vue.config.js:13]: Just curious, why is this needed?

**Review by gtmills — APPROVED** [review]

File: vue.config.js (hunk 1 above)
Comment (derick-montague) [reply vue.config.js:20]: Its true and could be a small performance boost.

File: vue.config.js (hunk 2 above)
Comment (derick-montague) [reply vue.config.js:13]: This is the gzip compression and we only want that for the build script. We also don't need the non-compressed files, so deleting deleteOringinalAssets removes the index.html, css, and js files from the dist directory.

**Review by yoshiemuranaka — APPROVED** [review]

**Comment by derick-montague** [comment]
Integrated into webui-vue in https://gerrit.openbmc-project.xyz/c/openbmc/webui-vue/+/28640
